java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java

```diff
@@ -456,4 +457,66 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
      * }</pre>
      */
     CompletableFuture<Object[]> zrankWithScore(String key, String member);
+
+    // TODO add @link to ZMPOP when implemented
+    /**
+     * Blocks the connection until it pops one member-score pair from the first non-empty sorted set
+     * in the provided list of key names.<br>
+     * <code>BZMPOP</code> is the blocking variant of <code>ZMPOP</code>.<br>
+     * A member with the highest/lowest score is popped from the first sorted set that is non-empty,
+     * with the given <code>keys</code> being checked in the order that they are given.<br>
+     * To pop more than one element use {@link #bzmpop(double, String[], ScoreModifier, long)}.
+     *
+     * @see <a href="https://redis.io/commands/bzmpop/">redis.io</a> for more details.
+     * @apiNote <code>BZMPOP</code> is a client blocking command, see <a
+     *     href="https://github.com/aws/glide-for-redis/wiki/General-Concepts#blocking-commands">Blocking
+     *     Commands</a> for more details and best practices.
+     * @param timeout The number of seconds to wait for a blocking operation to complete. A value of
+     *     <code>0</code> will block indefinitely.
+     * @param keys The keys of the sorted sets.
+     * @param modifier The element pop criteria.
+     * @return A two-element <code>array</code> containing the key name of the set from which an
+     *     element was popped, and a member-score <code>Map</code>.<br>
+     *     If no element could be popped and the timeout expired, returns </code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Object[] result = client.bzmpop(0.1, new String[] { "zSet1", "zSet2" }, MAX).get();
+     * Map<String, Double> data = (Map<String, Double>)result[1];
+     * String element = data.keySet().toArray(String[]::new)[0];
+     * System.out.printf("Popped '%s' with score %d from '%s'%n", element, data.get(element), result[0]);
+     * }</pre>
+     */
+    CompletableFuture<Object[]> bzmpop(double timeout, String[] keys, ScoreModifier modifier);
```

Comment:
Why is the return type a map but not for BZMPOPMAX? Seems confusing to have different return types for similar commands.

Reply:
Ask redis, not me